core/embed/bootloader/bootui.c

```diff
@@ -59,34 +51,28 @@
 #define COLOR_BL_GRAY COLOR_BL_FG
 #endif
 
-#define COLOR_WELCOME_BG COLOR_WHITE  // welcome background
-#define COLOR_WELCOME_FG COLOR_BLACK  // welcome foreground
-
 // common shared functions
 
-static void ui_confirm_cancel_buttons(void) {
-  display_bar_radius(9, 184, 108, 50, COLOR_BL_FAIL, COLOR_BL_BG, 4);
-  display_icon(9 + (108 - 16) / 2, 184 + (50 - 16) / 2, 16, 16,
-               toi_icon_cancel + 12, sizeof(toi_icon_cancel) - 12, COLOR_BL_BG,
-               COLOR_BL_FAIL);
-  display_bar_radius(123, 184, 108, 50, COLOR_BL_DONE, COLOR_BL_BG, 4);
-  display_icon(123 + (108 - 19) / 2, 184 + (50 - 16) / 2, 20, 16,
-               toi_icon_confirm + 12, sizeof(toi_icon_confirm) - 12,
-               COLOR_BL_BG, COLOR_BL_DONE);
+static void format_ver_bfr(const char *format, uint32_t version, char *bfr,
```

Comment:
united: https://github.com/trezor/trezor-firmware/commit/b5d385f8f8e25515eb410497730ac954d240d1b6

(besides returning it, you don't have to declare it every caller function. its allocated statically instead of on stack, which is a tradeoff. disadvantage is that you have to take care not to use it twice)